props/saber_chris_buttons.h

```diff
@@ -0,0 +1,1036 @@
+							You must exit COLOUR MENU to resume using lightsaber normally.
+
+							
+===========================================================		
+========================= DEFINES =========================
+#define FAKE_BLADE_ID
+Replaces regular BladeID and allows toggling between two different 
+blade/preset arrays regardless of actual BladeID status.
+
+#define SA22C_NO_LOCKUP_HOLD	
+Applicable to two-button mode only, reverts to lockup being triggered
+by clash while holding aux.
+
+GESTURE CONTROLS
+There are four gesture types: Twist, Stab, Swing and Thrust.
+Gesture controls bypass all preon effects.
+Below are the options to add to the config to enable the various gestures. MM.
+#define SA22C_TWIST_ON
+#define SA22C_TWIST_OFF
+#define SA22C_STAB_ON
+#define SA22C_SWING_ON
+#define SA22C_THRUST_ON
+#define SA22C_FORCE_PUSH
+
+#define SA22C_FORCE_PUSH_LENGTH 5
+Allows for adjustment to Push gesture length in millis needed to trigger Force Push
+Recommended range 1 ~ 10, 1 = shortest, easiest to trigger, 10 = longest. MM.
+
+Tightened click timings
+I've shortened the timeout for short and double click detection from 500ms
+to 300ms.  I think it feels more responsive this way but still gives enough
+timeout to ensure all button actions can be achieved consistently
+I've included all button timings so they can be easily tweaked to suit
+individual tastes. MM.
+
+============================================================
+============================================================
+*/
+#ifndef PROPS_SABER_SENSE_BUTTONS_H
+#define PROPS_SABER_SENSE_BUTTONS_H
+
+
+//  Used instead of BladeID and allows toggle switching 
+//  between two blade/preset arrays manually.
+#ifdef FAKE_BLADE_ID
+struct FakeBladeID {
+   static int return_value;
+   float id() { return return_value; }
+   static void toggle() {
+     return_value = NO_BLADE - return_value;
+   }
+};
+int FakeBladeID::return_value = 0;
+#undef BLADE_ID_CLASS_INTERNAL
+#define BLADE_ID_CLASS_INTERNAL FakeBladeID
+#undef BLADE_ID_CLASS
+#define BLADE_ID_CLASS FakeBladeID
+#endif
+
+
+#include "prop_base.h"
+#include "../sound/hybrid_font.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE SabersensePropV31
+
+#ifndef MOTION_TIMEOUT
+#define MOTION_TIMEOUT 60 * 15 * 1000
+#endif
+
+#ifndef SA22C_SWING_ON_SPEED
+#define SA22C_SWING_ON_SPEED 250
+#endif
+
+#ifndef SA22C_LOCKUP_DELAY
+#define SA22C_LOCKUP_DELAY 200
+#endif
+
+#ifndef SA22C_FORCE_PUSH_LENGTH
+#define SA22C_FORCE_PUSH_LENGTH 5
+#endif
+
+#ifndef BUTTON_DOUBLE_CLICK_TIMEOUT
+#define BUTTON_DOUBLE_CLICK_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_SHORT_CLICK_TIMEOUT
+#define BUTTON_SHORT_CLICK_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_HELD_TIMEOUT
+#define BUTTON_HELD_TIMEOUT 300
+#endif
+
+#ifndef BUTTON_HELD_MEDIUM_TIMEOUT
+#define BUTTON_HELD_MEDIUM_TIMEOUT 1000
+#endif
+
+#ifndef BUTTON_HELD_LONG_TIMEOUT
+#define BUTTON_HELD_LONG_TIMEOUT 2000
+#endif
+
+#ifdef SA22C_SWING_ON
+#define SWING_GESTURE
```

Comment:
Please use CHRIS_SWING_GESTURE instead of just SWING_GESTURE to avoid collisions.

Reply:
As above, all now changed to SABERSENSE.